Add Broker config `druid.broker.segment.watchRealtimeNodes` (#11732)
apache/druid: docs/configuration/index.md

|`druid.serverview.type`|batch or http|Segment discovery method to use. "http" enables discovering segments using HTTP instead of zookeeper.|batch|
|`druid.broker.segment.watchedTiers`|List of strings|Broker watches the segment announcements from processes serving segments to build cache of which process is serving which segments, this configuration allows to only consider segments being served from a whitelist of tiers. By default, Broker would consider all tiers. This can be used to partition your dataSources in specific Historical tiers and configure brokers in partitions so that they are only queryable for specific dataSources.|none|
|`druid.broker.segment.watchedDataSources`|List of strings|Broker watches the segment announcements from processes serving segments to build cache of which process is serving which segments, this configuration allows to only consider segments being served from a whitelist of dataSources. By default, Broker would consider all datasources. This can be used to configure brokers in partitions so that they are only queryable for specific dataSources.|none|

> |`druid.broker.segment.watchRealtimeNodes`|Boolean|Broker watches segment announcements from processes serving segments to build cache of which process is serving which segments. This configuration allows the exclusion of segments being served by realtime processes. By default, Broker would consider both realtime and historical processes. This can be used to configure brokers in partitions so that they are only queryable for Historical tiers or both Historical tiers and realtime processes.|true|

Reviewer: ```suggestion
|`druid.broker.segment.watchRealtimeNodes`|Boolean|The Broker watches segment announcements from processes that serve segments to build a cache to relate each process to the segments it serves.  When `watchRealtimeNodes` is true, the Broker watches for segment announcements from both Historicals and realtime processes. To configure a broker to exclude segments served by realtime processes, set `watchRealtimeNodes` to false. |true|
```
